stackslib/src/config/mod.rs

```diff
@@ -1210,53 +1210,321 @@ impl std::default::Default for Config {
 
 #[derive(Clone, Debug, Default, Deserialize, PartialEq)]
 pub struct BurnchainConfig {
+    /// The underlying blockchain used for Proof-of-Transfer.
+    /// Currently, only `"bitcoin"` is supported.
+    ///
+    /// Default: `"bitcoin"`
     pub chain: String,
+    /// The operational mode or network profile for the Stacks node.
+    /// This setting determines network parameters (like chain ID, peer version),
+    /// default configurations, genesis block definitions, and overall node behavior.
+    ///
+    /// Supported values: `"mainnet"`, `"mocknet"`, `"helium"`, `"neon"`, `"argon"`, `"krypton"`, `"xenon"`, `"nakamoto-neon"`.
+    ///
+    /// Default: `"mocknet"`
     pub mode: String,
+    /// The network-specific identifier used in P2P communication and database initialization .
+    /// Derived from `mode` during config load, unless explicitly overridden (for test purposes).
+    ///
+    /// Default: Derived from [`BurnchainConfig::mode`] ([`CHAIN_ID_MAINNET`] for `mainnet`, [`CHAIN_ID_TESTNET`] otherwise).
```

Comment:
I wonder if we should add a warning to several of these settings i.e. "Don't modify this unless you really know what you're doing"
chain_id is a great example of this.